### PR TITLE
Coerce all the static viz charts that could be a combo to be a combo

### PR DESCRIPTION
**Pull request by howonlee**
Pusuant to #24671. Previously there existed multiple backend render functions per time series, categorical line, area, bar charts. Now it's all `combo`. However, donuts, waterfall, and progress bar remain different backend render functions.

---

## Edit: By @adam-james-v 

Related to: https://github.com/metabase/metabase/issues/24759

This PR changes a few things:

1. remove categorical/timeline series line/area/bar Clojure code
2. Makes sure line/area/bar charts use the `js-svg/combo-chart` (this was actually already the case, or was changed in other static-viz PRs, so it's sufficient here).
3. remove sparkline defmethod, and sparkline namespaces (including the sparkline test namespace). There was a Slack discussion that sparkline is an older idea anyway, superceded by regular line chart renders. As well, the sparkline namespaces are marked as 'deprecated'.
4. Rewrite line/area/bar tests in `metabase.pulse.render.body-test` to use the combo renderer instead.

### Some follow-up work
I think some tests could be added to help with these sorts of changes. I want to merge this change, but have this PR: https://github.com/metabase/metabase/pull/25301 where I'm trying to add a few utility functions to help with static-viz rendering unit tests that work with data only, so that there is less need to manually render/check for edge cases.

### CODE REVIEW

**Comment by dpsutton** [comment]
Seems this removes calling the interface functions of
- "timeseries_bar"
- "timeseries_area"
- "timeseries_line"

Should these functions also be removed from `static_viz_interface.js`? And if so, do those files also have some component files that can be removed? like `frontend/src/metabase/static-viz/components/TimeSeriesLineChart` and all of the files there?

File: test/metabase/pulse/render/js_svg_test.clj [hunk 1]
```diff
@@ -91,33 +91,51 @@
...
-                                                     (re-matches #"prefix\d+\.\d{2}" %))))
-                          :x-axis-labels (s/+ (s/tuple
-                                               #{"#text"}
-                                               #(and (string? %)
-                                                     ;; ["#text" "1/1/2020"]
-                                                     (re-matches #"\d+/\d+/\d{4}" %)))))
-              text-nodes (->> svg-hiccup (tree-seq vector? rest) (filter text-node?))]
-          (is (= true (s/valid? spec text-nodes))
-              text-nodes))))))
+#_(defn goal-line-test []
```
Comment (dpsutton) [inline test/metabase/pulse/render/js_svg_test.clj:94]: is this intentionally commented out?

Comment (adam-james-v) [reply test/metabase/pulse/render/js_svg_test.clj:94]: That was a miss. Thanks!

**Comment by adam-james-v** [comment]
> 

That's a good point. They should be removable. @WiNloSt @alxnddr Could you confirm?

**Comment by dpsutton** [comment]
Checking trendlines

Before:
image

After:
image

Seems good. I was worried this one might not work out the same but seems good to me.

**Review by dpsutton — APPROVED** [review]
Seems good to me. I think there's lots of code that can still be deleted (and good call on deleting all of the old sparkline java 2d stuff)

**Comment by adam-james-v** [comment]
> Checking trendlines
>  
> Seems good. I was worried this one might not work out the same but seems good to me. I was nervous because the code didn't seem to care about `insights` any longer.

Trendlines pass the key `:smartscalar` on the card's `:display`. So, I simplified that condition by checking for that key instead, since a smartscalar card will have insights.

Lot's of code that can be deleted. Yep, that'll be handled in a follow up PR soon. Captured in the Epic!

**Pull request merged**